src/osemgrep/language_server/LS.ml

```diff
@@ -117,11 +117,22 @@ module MessageHandler = struct
           let content = Some first.text in
           scan_file ~content server uri;
           server
-      | CN.DidSaveTextDocument { textDocument = { uri }; _ }
-      | CN.TextDocumentDidOpen { textDocument = { uri; _ } } ->
-          Logs.app (fun m -> m "Scanning file %s" (Uri.to_string uri));
+      | CN.DidSaveTextDocument { textDocument = { uri }; _ } ->
+          Logs.app (fun m -> m "Scanning file %s on save" (Uri.to_string uri));
           scan_file server uri;
           server
+      | CN.TextDocumentDidOpen { textDocument = { uri; _ } } ->
+          let path = uri |> Uri.to_path |> Fpath.v in
+          let prev_scan = Session.previous_scan_of_file server.session path in
+          (* We usually scan every file on startup, so let's only rescan an opened
+             file if there weren't previous results *)
+          if Option.is_some prev_scan then
```

Comment:
usually more idiomatic to do a match prev_scan with Some _ -> ... | None -> ...

Reply:
Fair